src/java/org/apache/cassandra/cql3/SingleColumnRelation.java

```diff
@@ -275,6 +275,18 @@ protected Restriction newAnnRestriction(TableMetadata table, VariableSpecificati
         return new SingleColumnRestriction.AnnRestriction(columnDef, term);
     }
 
+    @Override
+    protected Restriction newAnalyzerMatchesRestriction(TableMetadata table, VariableSpecifications boundNames)
+    {
+        if (mapKey != null)
```

Comment:
Is this the right place for this check?

(maybe it is)

Reply:
This code is mostly copied from the `newLikeRestriction` in the nearby code. Looks like other methods don't have the check. I'll figure out what is correct.

Reply:
I think this is the correct place for the check. The method is called in the below code, and that block expects exceptions when the relation is invalid:

https://github.com/datastax/cassandra/blob/5093eedda666dbe2398e807131a9e3df0ab2618d/src/java/org/apache/cassandra/cql3/Relation.java#L131-L164